<?xml version="1.0"?>

<launch>
  <!-- Run box filter to remove car points -->
  <!-- <node pkg="lidar_snow_removal" type="cropBoxFilter" name="cropbox" output="screen"/>
    <param name="/cropbox/inputTopic" value="/velodyne_points" type="string" />
    <rosparam param="/cropbox/maxvector">[2.5, 0.935, 0]</rosparam>
    <rosparam param="/cropbox/minvector">[-2.5, -0.935, -1.480]</rosparam>
    <param name="/cropbox/setNegative" value="true" type="bool" />
    <param name="/cropbox/outputNoPoints" value="false" type="bool" /> -->
      <!-- minvector = [minX, minY, minZ] -->

  <!-- Run radius outlier removal filter
         - set the radius to search around each point
         - set the min. no. of neighbours to be within the search radius for the point to be included -->
  <node pkg="lidar_snow_removal" type="radiusOutlierFilter" name="radius_outlier_filter" output="screen"/>
    <param name="/radius/inputTopic" value="/velodyne_points" type="string" />
    <param name="/radius/radius_search" value="0.05" type="double" />
    <param name="/radius/minNeighbours" value="3" type="int" />

  <!-- Run dynamic radius outlier removal filter
        - set the min. no. of neighbours to be within the search radius for the point to be included
        - Set the radius search multiplier (i.e. radius_search = radius_multiplier * azAngle * range)
        - Set the azimuth angle of the lidar (in degrees, for VLP16 - 0.4)
        - Set the minimum search radius. This stops all the close points from being removed-->

  <node pkg="lidar_snow_removal" type="dynamicRadiusOutlierFilter" name="dynamic_radius_outlier_filter" output="screen"/>
    <param name="/DROR/inputTopic" value="/velodyne_points" type="string" />
    <param name="/DROR/min_Neighbours" value="3" type="int" />
    <param name="/DROR/radius_multiplier" value="3" type="double" />
    <param name="/DROR/azimuth_angle" value="0.16" type="double" />
    <param name="/DROR/min_search_radius" value="0.04" type="double" />
    <param name="/DROR/write_to_kitty_format" value="true" type="bool" />
    <param name="/DROR/output_directory" value="/home/nick/snow_removal_results" type="string" />

  <!-- Run statistical outlier removal filter
        - set the amount of neighbours to be analyzed for each point
        - set the standard deviation multiplier for the thresholding-->
  <node pkg="lidar_snow_removal" type="statsOutlierFilter" name="statistical_outlier_filter" output="screen"/>
    <param name="/stats/inputTopic" value="/velodyne_points" type="string" />
    <param name="/stats/meanK" value="50" type="double" />
    <param name="/stats/stdDev" value="0.3" type="double" />

  <!-- Run voxel grid filter -->
    <node pkg="lidar_snow_removal" type="voxelGridFilter" name="voxel_grid_filter" output="screen"/>
      <param name="/voxel/inputTopic" value="/velodyne_points" type="string" />
      <rosparam param="/voxel/leafSize">[0.2, 0.2, 0.2]</rosparam>

  <!-- Run rviz -->
    <node name="rviz1" pkg="rviz" type="rviz" args="-d $(find lidar_snow_removal)/config/rviz_config.rviz" />

</launch>
